django/contrib/contenttypes/fields.py

```diff
@@ -243,7 +241,8 @@ def __get__(self, instance, cls=None):
                 rel_obj = ct.get_object_for_this_type(pk=pk_val)
             except ObjectDoesNotExist:
                 pass
-        setattr(instance, self.cache_attr, rel_obj)
+
+        self.set_cached_value(instance=instance, value=rel_obj)
```

Comment:
In general, I think the `get` / `set` style APIs are familiar enough that we don't need to use keyword arguments with every call. But I don't feel strongly about it.